backend/remote-state/http/backend.go

```diff
@@ -0,0 +1,301 @@
+		unlockAddress := v.(string)
+		// Validate unlockAddress
+		validationErr := assertValidURL(unlockAddress)
+		if validationErr != nil {
+			return validationErr
+		}
+		b.unlockAddress = unlockAddress
+	} else {
+		// If unlockAddress is null, use the http rest api address
+		b.unlockAddress = b.address
+	}
+
+	b.unlockMethod = data.Get("unlock_method").(string)
+
+	if v, ok := data.GetOk("username"); ok {
+		b.username = v.(string)
+	}
+
+	if v, ok := data.GetOk("password"); ok {
+		b.password = v.(string)
+	}
+
+	client := &http.Client{
+		Timeout: time.Second * 10,
+	}
+	// skip_cert_verification client
+	if v, ok := data.GetOk("skip_cert_verification"); ok {
+		if b.skipTLS = v.(bool); b.skipTLS {
+			// If skip_cert_verification = true, the address must be of type HTTPS
+			if !isHTTPS(addressURL) {
+				return fmt.Errorf("Address must be of type HTTPS if skip_cert_verification = true")
+			}
+			// If local_ca_file is also set, raise an error
+			if v, ok := data.GetOk("local_ca_file"); ok {
+				return fmt.Errorf("skip_cert_verification is %t and local_ca_file is set: %s. please choose one or the other", b.skipTLS, v)
+			}
+			// If mutual_tls_authentication is also set, raise an error
+			if data.Get("mutual_tls_authentication").(bool) == true {
+				return fmt.Errorf("skip_cert_verification is true and mutual_tls_authentication is set. please choose one or the other")
+			}
+			log.Printf("[DEBUG] Using https client with skipping cert verification")
+			// add the option to ignores TLS verification to our client
+			client.Transport = &http.Transport{
+				TLSClientConfig: &tls.Config{
+					InsecureSkipVerify: true,
+				},
+			}
+		}
+	}
+	// CA provided client and mutual tls authentication client
+	if v, ok := data.GetOk("local_ca_file"); ok {
+		// If local_cert_ca_file exists, the address must be of type HTTPS
+		if !isHTTPS(addressURL) {
+			return fmt.Errorf("Address must be of type HTTPS if local_ca_file is set")
+		}
+		if data.Get("skip_cert_verification").(bool) == true {
+			return fmt.Errorf("skip_cert_verification is true and local_ca_file is set. Please choose one or the other")
+		}
+		b.localCertCAFile = v.(string)
+
+		// Get the SystemCertPool, continue with an empty pool on error
+		rootCAs, _ := x509.SystemCertPool()
+		if rootCAs == nil {
+			rootCAs = x509.NewCertPool()
+		}
+		// Read in the ca cert file
+		cert, err := ioutil.ReadFile(b.localCertCAFile)
+		if err != nil {
+			return fmt.Errorf("Failed to read %s into memory: %s", b.localCertCAFile, err)
+		}
+		// Append our cert to the system pool
+		if ok := rootCAs.AppendCertsFromPEM(cert); !ok {
+			return fmt.Errorf("No certs could be appended: %s", cert)
+		}
+		// add RootCas to our client
+		log.Printf("[DEBUG] Using https client with provided CA")
+		client.Transport = &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs: rootCAs,
+			},
+		}
+		if v, ok := data.GetOk("mutual_tls_authentication"); ok {
```

Comment:
`mutual_tls_authenication` is optional with a default value. you can safely use `d.Get()` instead of `d.GetOk()` here which should save you some indentation:

```
if b.mutualTLS = data.Get("mutual_tls_authentication").(bool): b.mutalTLS {
  // ...
}
```